InSource/TimerControl.cs

```diff
@@ -3,150 +3,179 @@
+		}
+
+		protected override void OnLoad(EventArgs e)
+		{
+			_timerSettings = _persistence.LoadRequest().Timer;
+
+			if (_timerSettings.HasTimer)
+			{
+				timerCheckBox.Checked = true;
+				refreshIntervalLbl.ForeColor = Color.Black;
+				SetIntervalTextBox(true, _timerSettings.Minutes.ToString());
+
+				_timeLeft = _timerSettings.Minutes;
+				Timer.Enabled = true;
+				remainingTimeLbl.Text = _timerSettings.Minutes + PluginResources.RemainingMinutes_Message;
+			}
+			else
+			{
+				timerCheckBox.Checked = false;
+				SetIntervalTextBox(false, @"0");
+
+				refreshIntervalLbl.ForeColor = Color.Gray;
+				SetRemainingLabel(PluginResources.DisabledTimer_Message, Color.Gray);
+
+				Timer.Enabled = false;
+
+			}
+			if (_persistence.LoadRequest().DeleteFolders)
+			{
+				deleteBtn.Checked = true;
+			}
+			else
+			{
+				archiveBtn.Checked = true;
+			}
+			archiveBtn.CheckedChanged += ArchiveBtn_CheckedChanged;
+
+		}
+
+		private void ArchiveBtn_CheckedChanged(object sender, EventArgs e)
+		{
+			bool deleteFolders;
+			if (!archiveBtn.Checked && deleteBtn.Checked)
+			{
+				deleteFolders = true;
+			}
+			else
+			{
+				deleteFolders = false;
+			}
+
+			_persistence.UpdateDelete(deleteFolders);
+		}
+
+		private void _timer_Elapsed(object sender, ElapsedEventArgs e)
+		{
+			if (_timeLeft > 0)
+			{
+				_timeLeft--;
+				remainingTimeLbl.Text = _timeLeft + @" minutes until project request is checked. ";
+			}
+			if (_timeLeft == 0)
+			{
+				OnCheckForProjectsRequestEvent();
+				_timeLeft = _timerSettings.Minutes;
+				remainingTimeLbl.Text = _timeLeft + PluginResources.RemainingMinutes_Message;
+			}
+		}
+
+		private void TimeTextBox_TextChanged(object sender, EventArgs e)
+		{
+			if (!string.IsNullOrWhiteSpace(intervalTextBox.Text) && Regex.IsMatch(intervalTextBox.Text, "^[0-9]*$"))
+			{
+				_timerSettings.Minutes = int.Parse(intervalTextBox.Text);
+
+				_persistence.SaveTimerSettings(_timerSettings);
+
+				_timeLeft = _timerSettings.Minutes;
+
+				remainingTimeLbl.Text = _timeLeft + PluginResources.RemainingMinutes_Message;
+				Timer.Enabled = true;
+			}
+			else
+			{
+				_messageBoxService.ShowWarningMessage(PluginResources.TimerNumberSetup_Message, string.Empty);
+			}
+		}
+
+		private void timerCheckBox_CheckedChanged(object sender, EventArgs e)
+		{
+			if (timerCheckBox.Checked)
+			{
+				intervalTextBox.Enabled = true;
+				refreshIntervalLbl.ForeColor = Color.Black;
+				remainingTimeLbl.ForeColor = Color.Black;
+				_timerSettings.HasTimer = true;
+				if(intervalTextBox.Text.Equals("0"))
+				{
+					remainingTimeLbl.Text = string.Empty;
+				}
+			}
+			else
+			{
+				SetIntervalTextBox(false, @"0");
+				refreshIntervalLbl.ForeColor = Color.Gray;
+				_timerSettings.HasTimer = false;
+				_timerSettings.Minutes = 0;
+
+				Timer.Enabled = false;
+				SetRemainingLabel(PluginResources.DisabledTimer_Message, Color.Gray);
+			}
+		}
+
+		private void SetIntervalTextBox(bool isEnabled, string text)
```

Comment:
This method can be refactored to receive a Label and it can be used for refreshIntervalLbl

Reply:
The changes can be checked in the new commit